src/senaite/patient/content/patient.py

```diff
@@ -314,18 +314,24 @@ def validate_mrn(data):
                 # nothing changed
                 return
 
-        if not patient_api.is_patient_required():
-            # MRN is not a required field
-            return
+        empty = not data.mrn
+        required = patient_api.is_patient_required()
 
-        if not data.mrn:
-            raise Invalid(_("Patient Medical Record is missing or empty"))
+        if empty and required:
+            # empty and required. It cannot be
+            raise Invalid(
+                _("invalid_patient_mrn_is_empty",
+                  default="Patient Medical Record is missing or empty"))
 
-        patient = patient_api.get_patient_by_mrn(
-            data.mrn, full_object=False, include_inactive=True)
+        elif empty:
+            # empty and non required
+            return
 
-        if patient:
-            raise Invalid(_("Patient Medical Record # must be unique"))
+        # search for uniqueness
+        if not patient_api.is_mrn_unique(data.mrn):
+            raise Invalid(
+                _("invalid_patient_mrn_non_unique",
+                  default="Patient Medical Record # must be unique"))
```

Comment:
I would rather write "Patient Medical Record Number must be unique" and above "Patient Medical Record Number is required"